x/staking/src/types/tx_proto.rs

```diff
@@ -183,6 +222,27 @@ pub struct Description {
 impl Protobuf<Description> for Description {}
 
 impl Description {
+    /// supplement_description updates the fields of a given description. An error is
+    /// returned if the resulting description contains an invalid length.
+    pub fn supplement_description(&self, other: &mut Description) -> Result<(), AppError> {
+        if other.moniker == DO_NOT_MODIFY_STRING {
```

Comment:
Maybe you should make them `Option<T>`?

Reply:
All comments about the feature. 
Yeah, I can use `Option<T>`. It'll improve the client side part. This fields form a struct with non-optional strings and then I need to setup `""` for the fields. The server side will not know where we need to change field and where we don't.  

I'll think about the issue. In this case some intermediate struct can help.

Reply:
I think I understand. You tried to use single struct in server and client parts?

Reply:
Yes, it's one structure. I'll change solution.

Reply:
updated